src/versionSocketConnection.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventEmitter } from 'events';
+import WebSocket from 'ws';
+import { WebSocketEvent } from './common/webviewEvents';
+
+export type IDevToolsPostMessageCallback = (e: WebSocketEvent, message?: string) => void;
+
+export interface BrowserVersionCdpResponse {
+   id: number;
+   result?: {
+       product?: string;
+       revision?: string;
+   }
+}
+
+export interface BrowserVersion {
+    major: number;
+    minor: number;
+    revision: string;
+}
+
+const MIN_SUPPORTED_BROWSER: BrowserVersion = {
+    major: 94,
+    minor: 975,
+    revision : '',
+};
+
+export class BrowserVerisonDetectionSocket extends EventEmitter {
+    private readonly targetUrl: string;
+    private socket: WebSocket | undefined;
+
+    constructor(targetUrl: string) {
+        super();
+        this.targetUrl = targetUrl;
+    }
+
+    dispose(): void {
+        if (this.socket) {
+            this.socket.close();
+            this.socket = undefined;
+        }
+    }
+
+    detectVersion(): void {
+        // Connect to target to determine browser version
+        this.socket = new WebSocket(this.targetUrl);
+        this.socket.onopen = () => this.onOpen();
+        this.socket.onmessage = ev => this.onMessage(ev);
+    }
+
+    private onOpen(): void {
+        // Send request to get browser version
+        const requestMessage = {
+            id: 0,
+            method: 'Browser.getVersion',
+            params: {},
+        };
+        if (this.socket) {
+            this.socket.send(JSON.stringify(requestMessage));
+        }
+    }
+
+    private onMessage(message: { data: WebSocket.Data }) {
+        // Determine if this is the browser.getVersion response and send revision hash to devtoolsPanel
+        const data = JSON.parse(message.data.toString()) as BrowserVersionCdpResponse;
+        this.emit('setBrowserRevision', this.calcBrowserRevision(data));
+        // Dispose socket after version is determined
+        this.dispose();
+        return;
+    }
+
+    private calcBrowserRevision(data: BrowserVersionCdpResponse): string {
+        if (data.id !== 0 || !data.result || !data.result.product && !data.result.revision) {
+            return '';
+        }
+        // product in the form Edg/Major.0.Minor.0 or HeadlessEdg/Major.0.Minor.0
```

Comment:
I'm not 100% sure this is true. Anyone know if this is accurate?

Reply:
Poking around I've at least seen the last number not be zero. Probably best to err on the safe side and assume all can be non-zero (assuming you don't actually need to consume `major` or `minor` from other parts of the code).

```ts
const minSupported = '94.0.975.0'.split('.');
// ...
return parts.some((v, i) => v < minSupported[i]) ? '' : currVersion.revision;
```

Reply:
Updated

Reply:
Well, perhaps don't use my incorrect code, but the idea stands 😀